### PR TITLE
Adding functions uri in environment variables in FunctionsNetHost

### LINKED ISSUE #2121: The gRPC channel URI 'http://:' could not be parsed.
Azure function worked yesterday with zero issues.  I made a small change to add a new model field to my function code, published, and now I only get this exception.

#### Investigative information

Please provide the following:

- Timestamp: 2023-11-17T20:28:24Z
- Function App version: ~4
- Function App name: SCMessageProcessor
- Function name(s) (as appropriate):  Function1
- Invocation ID: 4f17817d-021f-4a78-b5c8-b5b3732118c0
- Region: South Central US 

Works fine when running locally but not after any publish.
Worked yesterday

#### Repro steps

I replay a message from my queue and I see the invocation error in the logs.

#### Expected behavior
When a new message is queued, the function should invoke without error and process the message like it did yesterday.

#### Actual behavior
https://gist.github.com/wheelie33/10625a57637aca4de1ba9da219f63404

#### Related information 

host.json:
```
{
    "version": "2.0",
  "logging": {
    "logLevel": { "default": "Information" },
    "fileLoggingMode": "always",
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      },
      "enableLiveMetricsFilters": true
    }
  }
}
```

program.cs
```
using AzureMessageFunctionsEf.Data;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Logging;


using var loggerFactory = LoggerFactory.Create(builder =>
{
    builder.AddConsole();
});

var logger = loggerFactory.CreateLogger<Program>();


try
{
    var config = new ConfigurationBuilder()
            .AddJsonFile("local.settings.json", optional: true)
            .AddEnvironmentVariables()
            .Build();


    var connString = config.GetConnectionString("Default");

   

    var host = new HostBuilder()
        .ConfigureAppConfiguration(builder => {
            builder.AddEnvironmentVariables();
            builder.AddJsonFile("local.settings.json", optional: true);


        })
            .ConfigureServices(services =>
            {

                services.AddDbContext<DatabaseContext>(options => options.UseSqlServer(connString, options =>
                {
                    options.EnableRetryOnFailure();
                }));
            })
        .ConfigureFunctionsWorkerDefaults()
        .Build();

    await host.RunAsync();
}
catch(Exception ex)
{
    logger.LogCritical("JW: " + ex.Message);
}

```

Packages:

```
<ItemGroup>
    <PackageReference Include="Azure.Communication.Email" Version="1.0.1" />
    <PackageReference Include="Azure.Communication.Sms" Version="1.0.1" />
    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.17.0" />
    <PackageReference Include="Azure.Storage.Blobs" Version="12.19.1" />
    <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.20.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.ServiceBus" Version="5.14.1" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.16.2" />
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.13" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="7.0.13" />
  </ItemGroup>
  <ItemGroup>
```

Azure Configuration
```
[
  {
    "name": "APPLICATIONINSIGHTS_CONNECTION_STRING",
    "value": "InstrumentationKey=06acf3bf-efda-4b61-ba72-760374208516;IngestionEndpoint=https://southcentralus-3.in.applicationinsights.azure.com/;LiveEndpoint=https://southcentralus.livediagnostics.monitor.azure.com/",
    "slotSetting": false
  },
  {
    "name": "AzureWebJobs.Function1.Disabled",
    "value": "0",
    "slotSetting": false
  },
  {
    "name": "AzureWebJobsStorage",
    "value": "DefaultEndpointsProtocol=https;AccountName=scmessageworkerfunctions;AccountKey={REDACTED};EndpointSuffix=core.windows.net",
    "slotSetting": false
  },
  {
    "name": "BlobConnectionString",
    "value": "DefaultEndpointsProtocol=https;AccountName=scoutconnectstore;AccountKey={REDACTED};EndpointSuffix=core.windows.net",
    "slotSetting": false
  },
  {
    "name": "CommunicationConnectionString",
    "value": "endpoint=https://sc-production-comm-service.unitedstates.communication.azure.com/;accesskey={REDACTED}",
    "slotSetting": false
  },
  {
    "name": "FromEmail",
    "value": "{REDACTED}",
    "slotSetting": false
  },
  {
    "name": "FromNumber",
    "value": "{REDACTED}",
    "slotSetting": false
  },
  {
    "name": "FUNCTIONS_EXTENSION_VERSION",
    "value": "~4",
    "slotSetting": false
  },
  {
    "name": "FUNCTIONS_WORKER_RUNTIME",
    "value": "dotnet-isolated",
    "slotSetting": false
  },
  {
    "name": "SCM_DO_BUILD_DURING_DEPLOYMENT",
    "value": "0",
    "slotSetting": false
  },
  {
    "name": "ServiceBusConnectionString",
    "value": "Endpoint=sb://scoutconnect.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey={REDACTED}",
    "slotSetting": false
  },
  {
    "name": "WEBSITE_USE_PLACEHOLDER_DOTNETISOLATED",
    "value": "1",
    "slotSetting": false
  }
]```

### PR DESCRIPTION
**Pull request by surgupta-msft**
### Issue describing the changes in this PR

resolves #2121

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: host/src/FunctionsNetHost/Grpc/IncomingGrpcMessageHandler.cs [hunk 1]
```diff
@@ -10,10 +10,13 @@ internal sealed class IncomingGrpcMessageHandler
     {
         private bool _specializationDone;
         private readonly AppLoader _appLoader;
+        private readonly GrpcWorkerStartupOptions _grpcWorkerStartupOptions;
+        private const string HostEndpoint = "Functions:Worker:HostEndpoint";
```
Comment (kshyju) [inline host/src/FunctionsNetHost/Grpc/IncomingGrpcMessageHandler.cs:14]: We have a class `EnvironmentVariables` to store environment variable keys. Let's move this there.

**Comment by kshyju** [comment]
We should also bump the version [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/b872d4ed6d42b052edf582ac476890add4148015/host/tools/build/Microsoft.Azure.Functions.DotnetIsolatedNativeHost.nuspec#L7)
Once merged and package is published, we should update host to consume this version. Example PR - https://github.com/Azure/azure-functions-host/pull/9740

**Review by kshyju — APPROVED** [review]

**Comment by Wangor** [comment]
Seems promising, is it in the next version or just to update the current host package? Thanks a lot!

**Comment by kshyju** [comment]
> Seems promising, is it in the next version or just to update the current host package? Thanks a lot!

This will be in the next host update.

**Comment by Wangor** [comment]
sorry for the question. I installed the Azure function tools for local debugging. This seems to have still the old workers and hosts. Any timeline when it is updated or can i download the zip and replace some stuff? Thanks a lot for your work!

**Comment by kshyju** [comment]
Once the host/core tools version with this fix is released, Visual studio will automatically pull in that version for you. Hopefully this will be out in the second half of January.

**Comment by Wangor** [comment]
Thanks. That means Isolated Functions are officially not working on .NET / Azure as long it is not released?
(No clue if it works on azure but makes no sense to deploy untested code? :) )

**Comment by kshyju** [comment]
> Thanks. That means Isolated Functions are officially not working on .NET / Azure as long it is not released? (No clue if it works on azure but makes no sense to deploy untested code? :) )

Isolated functions will work with the existing features. Any new features we ship will be working only after it is officially released.

If you are experiencing issues when using dotnet isolated functions. please open a new GH issue with relevant details. We are happy to help.

**Comment by Wangor** [comment]
Now you make me curious... What i did: Startes Visual Studio, created a isolated function project with a simple httptrigger or scheduled trigger and both did not work at all. In debugging there was the issue that the function host returned a corrupt url. like http://:1234 which seemed to match somehow your fix...?

**Pull request merged**